### PR TITLE
=str #16973: Stream TCP actor should watch IO actor and handle termiation

### LINKED ISSUE #16973: Watch the connection actor from the Stream actor

### PR DESCRIPTION
**Pull request by drewhk**
Fixes #16973

### CODE REVIEW

**Comment by rkuhn** [comment]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala [hunk 1]
```diff
@@ -380,6 +381,23 @@ class TcpSpec extends AkkaSpec("akka.io.tcp.windows-connection-abort-workaround-
...
+    "handle when connection actor terminates unexpectedly" in {
+      import system.dispatcher
+
+      val serverAddress = temporaryServerAddress()
+      val binding = Tcp().bindAndHandle(Flow[ByteString], serverAddress.getHostName, serverAddress.getPort)
+
+      val result = Source.lazyEmpty[ByteString].via(Tcp().outgoingConnection(serverAddress)).runFold(0)(_ + _.size)
+
+      // Getting rid of existing connection actors by using a blunt instrument
+      system.actorSelection(akka.io.Tcp(system).getManager.path / "selectors" / "$a" / "*") ! Kill
```
Comment (rkuhn) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: Ah, forgot to say: this should be done in a separate ActorSystem if we ever run test cases in parallel.

Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: well, can't you annotate tests to not run in parallel?

Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: ah, I just add a system in scope, that is what you meant.

**Pull request merged**